parlai/tasks/mnist_qa/agents.py

```diff
@@ -0,0 +1,66 @@
+# Copyright (c) 2017-present, Facebook, Inc.
+# All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree. An additional grant
+# of patent rights can be found in the PATENTS file in the same directory.
+
+"""
+This is a simple question answering task on the MNIST dataset.
+In each episode, agents are presented with a number, which they are asked to
+identify.
+
+Useful for debugging and checking that one's image model is up and running.
+"""
+
+from parlai.core.dialog_teacher import DialogTeacher
+from .build import build
+
+import json
+import os
+
+
+def _path(opt):
+    build(opt)
+    dt = opt['datatype'].split(':')[0]
+
+    labels_path = os.path.join(opt['datapath'], 'mnist', dt, 'labels.json')
+    image_path = os.path.join(opt['datapath'], 'mnist', dt)
+    return labels_path, image_path
+
+
+class MnistQATeacher(DialogTeacher):
+    """
+    This version of MNIST inherits from the core Dialog Teacher, which just
+    requires it to define an iterator over its data `setup_data` in order to
+    inherit basic metrics, a `act` function, and enables
+    Hogwild training with shared memory with no extra work.
+    """
+    def __init__(self, opt, shared=None):
+        self.datatype = opt['datatype']
```

Comment:
actually the split function from before is still needed: it converts e.g. train:ordered to train